data/txHandler.go

```diff
@@ -777,21 +777,19 @@ type validatedIncomingTxMessage struct {
 
 // validateIncomingTxMessage is the validator for the MessageProcessor implementation used by P2PNetwork.
 func (handler *TxHandler) validateIncomingTxMessage(rawmsg network.IncomingMessage) network.ValidatedMessage {
-	msgKey, capguard, shouldDrop := handler.incomingMsgDupErlCheck(rawmsg.Data, rawmsg.Sender)
+	var msgKey *crypto.Digest
+	var isDup bool
 
-	accepted := false
-	defer func() {
-		// if we failed to put the item onto the backlog, we should release the capacity if any
-		if !accepted && capguard != nil {
-			if capErr := capguard.Release(); capErr != nil {
-				logging.Base().Warnf("validateIncomingTxMessage: failed to release capacity to ElasticRateLimiter: %v", capErr)
-			}
-		}
-	}()
+	if handler.msgCache != nil {
+		// check for duplicate messages
+		// this helps against relaying duplicates
 
-	if shouldDrop {
-		// this TX message was found in the duplicate cache, or ERL rate-limited it
-		return network.ValidatedMessage{Action: network.Ignore, ValidatedMessage: nil}
+		// note, this is the same check as in incomingMsgDupErlCheck prologue
+		// but for p2p network handlers no ERL needed because libp2p has own congestion control
```

Comment:
IMO it does not make much sense - a new function would be a single call inside two if-conds by cost of adding extra lines in caller to check return codes.

> the two halves don't have any relation to each other AFAICT

they do (both signal if a messages should be rejected with the same code) and they don't since use different message traits to decide on.

Reply:
Semi-orthogonal, do we understand the rate limiting behavior of libp2p vs our own implementation? It's reasonable to want to use that, but we should at least understand how they each perform relative to one another no?